### DISCUSSION TITLE
UnknownHostException <S3bucketname>.<kubernetesServicename>

### DISCUSSION THREAD

**Discussion by nareshurs** [minio/minio#13426] [Q&A] [answered]
Hello,

We are starting to use Minio as gateway to communicate with AWS S3 using Kubernetes. I have added Minio via Bitnami helm chart. I am trying to read data from S3 with spark end point url as 

sparkConf.set("spark.hadoop.fs.s3a.endpoint", minioK8sServiceName:9000");

However when I am running the code, It throws java.net.UnknownHostException: <bucketname>.<minioK8sServiceName>

Does Minio evaluate host name as bucketname.minioK8sServiceName If yes do we need to configure anything to resove DNS host?  Am I missing anything? Standalone code works fine but when we integrate with our cluster above error comes up

**Comment by harshavardhana** [comment, marked as answer]
> We are starting to use Minio as gateway to communicate with AWS S3 using Kubernetes. I have added Minio via Bitnami helm chart. I am trying to read data from S3 with spark end point url as

We do not support bitnami helm chart @nareshurs - you may ask Bitnami guys. 

> sparkConf.set("spark.hadoop.fs.s3a.endpoint", minioK8sServiceName:9000");
> 
> However when I am running the code, It throws java.net.UnknownHostException: .

You need to also set ```("spark.hadoop.fs.s3a.path.style", "true")```
